src/main/scala/cognite/spark/v1/AssetsHierarchyBuilder.scala

```diff
@@ -0,0 +1,230 @@
+package cognite.spark.v1
+
+import cats.effect.IO
+import cats.implicits._
+import cognite.spark.v1.SparkSchemaHelper.{fromRow, structType}
+import com.cognite.sdk.scala.common.SetValue
+import com.cognite.sdk.scala.v1.{Asset, AssetCreate, AssetUpdate, AssetsFilter, CogniteExternalId}
+import org.apache.spark.sql.types.{DataTypes, StructType}
+import org.apache.spark.sql.{DataFrame, SQLContext}
+
+final case class AssetsIngestSchema(
+    externalId: String,
+    parentExternalId: String,
+    source: Option[String],
+    name: String,
+    description: Option[String],
+    metadata: Option[Map[String, String]])
+
+object AssetsIngestSchema {
+  def toAssetCreate(a: AssetsIngestSchema): AssetCreate =
+    AssetCreate(
+      name = a.name,
+      description = a.description,
+      externalId = Some(a.externalId),
+      metadata = a.metadata,
+      source = a.source,
+      parentExternalId = Some(a.parentExternalId)
+    )
+
+  def toAssetUpdate(a: AssetsIngestSchema): AssetUpdate =
+    AssetUpdate(
+      name = Some(SetValue(a.name)),
+      description = a.description.map(SetValue(_)),
+      externalId = Some(SetValue(a.externalId)),
+      metadata = a.metadata.map(SetValue(_)),
+      source = a.source.map(SetValue(_)),
+      parentExternalId = Some(SetValue(a.parentExternalId))
+    )
+}
+
+final case class MultipleRootsException(message: String = "Tree has more than one root.")
+    extends Exception(message)
+final case class NoRootException(
+    message: String = """Tree has no root. Set parentExternalId to "" for the root Asset.""")
+    extends Exception(message)
+final case class InvalidTreeException(message: String = s"The tree is has an invalid structure.")
+    extends Exception(message)
+
+class AssetsHierarchyBuilder(config: RelationConfig)(val sqlContext: SQLContext)
+    extends CdfRelation(config, "assetshierarchybuilder") {
```

Comment:
Shouldn't this also be `"assetshierarchy"`?

Reply:
I should!